src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -140,10 +140,28 @@ private static boolean isInstancePrimitive(@Nonnull Class<?> type, @Nonnull Obje
         return findMatchingMethod(receiver, method, args);
     }
 
+    private static boolean isEnumInitializer(Method m) {
+        if (!m.getDeclaringClass().isEnum()) {
+            return false;
+        }
+        if (!m.getName().equals("$INIT")) {
```

Comment:
Hmm, is this name guaranteed by the JLS? Not a real problem if it is not, just wondering if we can make this a bit more portable.